### DISCUSSION TITLE
keyball39 3-7 ファームウェアFLASH時タイムアウトエラー

### DISCUSSION THREAD

**Discussion by tkhnct** [Yowkees/keyball#375] [Q&A] [answered]
keyball39をビルド中です。
ビルドガイド3-7の動確にあたり、
"REMAPに登録してあるKeyball_test.hex を書き込み(FLASH)してください。"
を実施したところ、以下タイムアウトエラーが表示されます。

> Reading the firmware binary from the server.
> Reading the firmware binary done.
> Initialize a bootloader.
> Fetching the Software Identifier.
> Caterina bootloader is not detected.
> Writing the firmware finished successfully.
> Error: Reading bytes was timeout.

エラーの為、ピンセットショートによる確認には反応がありません。
Mac12.6 + Chromeの構成となります。考えられる原因・対策があればご教示いただけますと幸いです。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

ProMicroをKeyballから取り外して、ProMicro単体での書き込みでも書き込みできないでしょうか。
その場合、ProMicroの不良と思われます。
また、REMAPでブートローダーを別のものに切り替えられるので別のものでも書き込みを試していただければ幸いです。

以上、お手数ですがご確認お願いいたします。

**Reply by tkhnct** [reply]
ありがとうございます。
結論ProMicro不良が疑われる為、別の物を購入して再度試し、完了後にチケットクローズ連絡を行います。

PC変更→変わらず
ProMicro単体→変わらず
ブートローダ変更→変わらず
QMK→デバイス認識せず
スペック高いUSBケーブルではProMicroのLED点灯せず
安いコードで反応するが上記反応せず

**Reply by tkhnct** [reply]
遊舎工房のKeyball委託販売ページにてリンクされているProMicroを買い直し、
ProMicro単体で接続を試してみたのですが解消されず、
ProMicro計5つ試してもうまくいかない為、ProMicroの不良の可能性は低いように思います。

caterina
> Reading the firmware binary from the server.
> Reading the firmware binary done.
> Initialize a bootloader.
> Fetching the Software Identifier.
> Caterina bootloader is not detected.
> Writing the firmware finished successfully.
> Error: Reading bytes was timeout.

dfu
> Reading the firmware binary from the server.
> Reading the firmware binary done.
> Device detected: Vendor ID:9025, Product ID:32822
> Error: Unsupported device: Vendor ID: 2341 Product ID: 8036

改めてで大変申し訳ないのですが、PC・ケーブル・ブラウザ・セキュリティなど
どこかに原因があるのではと思うのですが、確認すべき観点がないかご教示いただけますと大変ありがたいです。

**Comment by Yowkees** [comment]
ProMicroをすでに何個かお試しいただいたということで、確かに初期不良の可能性は低く、動作環境によるものと思います。
ただ私もProMicroの仕様を完璧に理解しているわけではないため、大変申し訳ありませんがProMicroのご購入先へお問合せいただければ幸いです。
お手数ですがご検討よろしくお願いいたします。

**Comment by tkhnct** [comment, marked as answer]
こちらご対応いただきまして、ありがとうございました。
遊舎工房様にお問い合わせを行いまして、2点の修正で書き込みできました。
1 QMK Toolboxを使用する
2 USBケーブルを変える

2について、USBケーブルによってProMicroに通電されるか挙動が異なり、
USB-C→USB-A→USB-Cと変換コネクタを何度か経由するとなぜか上手くいきました。
この原因はわかりかねる為、直接接続可能なケーブルは模索してまいります。
この度はご対応いただきまして、ありがとうございました。
